fetch/cosmwasm_contract/src/contract.rs

```diff
@@ -487,8 +513,12 @@ fn try_deposit<S: Storage, A: Api, Q: Querier>(
     let env_message_sender = deps.api.human_address(&env.message.sender)?;
 
     let amount = amount_from_funds(&env.message.sent_funds, state.denom.clone())?;
+    let increased_supply = state.supply + amount;
+    if increased_supply > state.cap {
+        return Err(StdError::generic_err(ERR_CAP_EXCEEDED));
+    }
```

Comment:
This is not supposed to be there. Depositr is adminisrative action - we can do whatever we want.
The `cap` is only for public API - it limits only `swap(...)` calls.

Reply:
I see that it is in ethereum contract - we need to think about this. Now is thew chance to change things.
But I think that this implementation is not desired = current ethereum contract impl. is incorrect.

Reply:
I am adding it to be symmetrical with your implementation https://github.com/fetchai/fetch-ethereum-bridge-v1/blob/master/ethereum/contracts/Bridge.sol#L650, unless you dropped the check in you latest pr

Reply:
To make my point, we can look at this from the perspective of changing `cap` value via `setCap(...)` call, which allows to **reduce** the `cap` value => it is admin action & it allows to create situation where `cap < supply`, what would be the same outcome is this admin action (= `deposit(...)`) would increate `suply` value over the `cap` limit.

Reply:
The same situation is with minting (available only in ethereum contract).

Technically speaking, the implementation with this check is correct in the sense that it will work and not cause fatal issues. It is just about loosing degree of freedom going freely over `cap` while depositing new supply **and at the same time** keeping `cap` effective limiting pubic API.

Reply:
I ditched this behaviour on ethereum side - [see this commit in the PR #40](https://github.com/fetchai/fetch-ethereum-bridge-v1/pull/40/commits/d7b1856c4e6ad729899698246efd24e70423c4c7)

Reply:
ditched it as well